django_lightweight_queue/management/commands/queue_configuration.py

```diff
@@ -1,46 +1,14 @@
-import warnings
 from typing import Any
 
-from django.core.management.base import BaseCommand, CommandParser
-
 from ... import app_settings
-from ...utils import get_backend, get_queue_counts, load_extra_settings
-from ...constants import SETTING_NAME_PREFIX
+from ...utils import get_backend, get_queue_counts
+from ...command_utils import CommandWithExtraSettings
 from ...cron_scheduler import get_cron_config
 
 
-class Command(BaseCommand):
-    def add_arguments(self, parser: CommandParser) -> None:
-        extra_settings_group = parser.add_mutually_exclusive_group()
-        extra_settings_group.add_argument(
-            '--config',
-            action='store',
-            default=None,
-            help="The path to an additional django-style config file to load "
-                 "(this spelling is deprecated in favour of '--extra-settings')",
-        )
-        extra_settings_group.add_argument(
-            '--extra-settings',
-            action='store',
-            default=None,
-            help="The path to an additional django-style settings file to load. "
-                 f"{SETTING_NAME_PREFIX}* settings discovered in this file will "
-                 "override those from the default Django settings.",
-        )
-
+class Command(CommandWithExtraSettings):
     def handle(self, **options: Any) -> None:
-        extra_config = options.pop('config')
-        if extra_config is not None:
-            warnings.warn(
-                "Use of '--config' is deprecated in favour of '--extra-settings'.",
-                category=DeprecationWarning,
-            )
-            options['extra_settings'] = extra_config
-
-        # Configuration overrides
-        extra_settings = options['extra_settings']
-        if extra_settings is not None:
-            load_extra_settings(extra_settings)
+        super().handle_extra_settings(**options)
```

Comment:
Have you considered just making this part of `handle` on the super class and then having subclasses call `super().handle(...)` instead of a custom method?

I can't see any weird side effects and it's probably a bit more expected when going through inheritance.

Reply:
I did, unfortunately Django's `BaseCommand.handle` raises `NotImplementedError` so we'd need where to stop the `super()` calls to avoid errors and also means that calling `super().handle(...)` in an implementation could easily look like it's a bug.
I also considered an approach of adding this functionality as a decorator (which would make it harder to get wrong) however that felt more complicated (class decorators being more complicated than function ones) and likely wouldn't have a great way to support the return value.

Reply:
Ah right gotcha.